internal/librarian/tag_and_release.go

```diff
@@ -157,6 +159,18 @@ func (r *tagAndReleaseRunner) processPullRequest(ctx context.Context, p *github.
 	return r.replacePendingLabel(ctx, p)
 }
 
+func determineTagFormat(libraryID string, librarianState *config.LibrarianState) (string, error) {
+	// TODO(#2177): read from LibrarianConfig
+	libraryState := librarianState.LibraryByID(libraryID)
+	if libraryState == nil {
+		return "", fmt.Errorf("library %s not found", libraryID)
+	}
+	if libraryState.TagFormat != "" {
+		return libraryState.TagFormat, nil
+	}
+	return "", fmt.Errorf("library %s did not configure tag_format", libraryID)
```

Comment:
I don't think tag format can be empty because we validate the state in https://github.com/googleapis/librarian/blob/c61df5eb73098dac9d117677c4846a0bc432fcbf/internal/librarian/state.go#L89